### PR TITLE
PAY-5702: Adds docs for the updated vault feature in Payment Services

### PR DESCRIPTION
**Pull request by danidelcar**
## Purpose of this pull request

This pull request (PR) adds docs for the updated vault feature in Payment Services. Created a new vault folder in the GraphQL schema with new queries and mutations.

## Affected pages

- ...

## Links to Magento Open Source code

- ...

## Additional info

[PAY-5702](https://jira.corp.adobe.com/browse/PAY-5702)

### CODE REVIEW

**Comment by danidelcar** [comment]
PR still in draft mode because I'm still missing some input/output attribute details to be clarified

File: src/pages/graphql/schema/checkout/index.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ title: Checkout
 
 The checkout schema helps complete the customer checkout process by managing and retrieving payment tokens and checkout agreements.
 
-The checkout workflow differs for each online payment time. See the following topics for more information.
+The checkout workflow differs for each online payment type. See the following topics for more information.
```
Comment (virfaure) [inline src/pages/graphql/schema/checkout/index.md:9]: Good catch 😆

File: src/pages/graphql/schema/vault/index.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+<InlineAlert variant="info" slots="text" />
+
+This query is available only if you have installed [Payment Services for Adobe Commerce](https://commercemarketplace.adobe.com/magento-payment-services.html) 2.10.0 or higher.
+
+The vault schema provides GraphQL information to generate **vault** (stored payment details) information endpoints. This schema also provides mutations for modifying a payment token.
+
+Once vaulted, a payment method can be reused as many times as possible without your customer needing to re-enter or authenticate their payment information.
+
+These endpoints can be used to vault stored payment details without a purchase, but tt is possible to [vault a payment method during checkout](https://developer.adobe.com/commerce/webapi/graphql/payment-services/vault/), see the [`customer-payment-tokens`](https://developer.adobe.com/commerce/webapi/graphql/schema/checkout/queries/customer-payment-tokens/) query for more information.
```
[the remark this answers is not in the document]
Comment (danidelcar) [reply src/pages/graphql/schema/vault/index.md:15]: ooops, good catch

File: src/pages/graphql/schema/vault/index.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
+title: Vault
+---
+
+# Vault
+
+<InlineAlert variant="info" slots="text" />
+
+This query is available only if you have installed [Payment Services for Adobe Commerce](https://commercemarketplace.adobe.com/magento-payment-services.html) 2.10.0 or higher.
```
[the remark this answers is not in the document]
Comment (Tomasito665) [reply src/pages/graphql/schema/vault/index.md]: I disagree with this change. There is no such thing as "endpoints" in GraphQL. There is only one endpoint: `/graphql`. Operations are then defined as `queries` and `mutations`, not endpoints.

[the remark this answers is not in the document]
Comment (Tomasito665) [reply src/pages/graphql/schema/vault/index.md]: I disagree - see https://github.com/AdobeDocs/commerce-webapi/pull/386/files/bf1fd65df382d19f6180afbce172eb6fe69ff091#r1886838346.

We should call it `These queries`, `These mutations`, or `These queries and mutations`, not `These endpoints`, as there is only one endpoint, `/graphql`.

File: src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        "createVaultCardSetupToken": {
+            "setup_token": "11M29729J2618171X"
+        }
+    }
+}
+```
+
+## Input attributes
+
+The `createVaultCardSetupToken` mutation must contain the following input attributes:
```
Comment (keharper) [inline src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md:76]:
```suggestion
The `CreateVaultCardSetupTokenInput` object must contain the following attributes:
```

File: src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`setup_token` | [VaultSetupTokenInput]! | The setup token information.
+`three_ds_mode` | Boolean | Indicates which 3D Secure authentication mode is in use. The possible values are `OFF`, `SCA_WHEN_REQUIRED`, `SCA_ALWAYS`.
```
Comment (keharper) [inline src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md:81]:
```suggestion
`three_ds_mode` | ThreeDSMode | Indicates which 3D Secure authentication mode is in use. The possible values are `OFF`, `SCA_WHEN_REQUIRED`, `SCA_ALWAYS`.
```

File: src/pages/graphql/schema/vault/mutations/createVaultCardPaymentToken.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+`card_description` | Boolean | The description of the vaulted card.
+
+## Output attributes
+
+The `CreateVaultCardPaymentTokenOutput` object contains the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`vault_token_id` | String! | The vault payment token information.
+`payment_source` | [PaymentSourceOutput]! | The payment source information.
```
Comment (keharper) [inline src/pages/graphql/schema/vault/mutations/createVaultCardPaymentToken.md:86]:
```suggestion
`payment_source` | PaymentSourceOutput! | The payment source information.
```
You are pretty consistently inserting braces (`[` `]`) where they don't belong. The braces indicate an array. I'm trying to correct them all, but double check.

File: src/pages/graphql/schema/vault/mutations/createVaultCardPaymentToken.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+### `PaymentSourceOutput` attributes
+
+The `PaymentSourceOutput` object describes the payment source information. Requires the following attribute:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`card` | [CardPaymentSourceOutput]! | The card payment source information.
```
Comment (keharper) [inline src/pages/graphql/schema/vault/mutations/createVaultCardPaymentToken.md:94]:
```suggestion
`card` | CardPaymentSourceOutput! | The card payment source information.
```

File: src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+}
+```
+
+## Input attributes
+
+The `createVaultCardSetupToken` mutation must contain the following input attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`setup_token` | [VaultSetupTokenInput]! | The setup token information.
```
Comment (keharper) [inline src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md:80]:
```suggestion
`setup_token` | VaultSetupTokenInput! | The setup token information.
```

File: src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+`three_ds_mode` | Boolean | Indicates which 3D Secure authentication mode is in use. The possible values are `OFF`, `SCA_WHEN_REQUIRED`, `SCA_ALWAYS`.
+
+### `VaultSetupTokenInput` attribute
+
+The `VaultSetupTokenInput` object describes the variables needed to create a vault card setup token. Requires the following attribute:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`payment_source` | [PaymentSourceInput]! | The payment source of the payment method.
```
Comment (keharper) [inline src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md:89]:
```suggestion
`payment_source` | PaymentSourceInput! | The payment source of the payment method.
```

File: src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+### `PaymentSourceInput` attribute
+
+The `PaymentSourceInput` object describes the payment source information of the payment method.
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`card` | [CardPaymentSourceInput]! | The card payment source information.
```
Comment (keharper) [inline src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md:97]:
```suggestion
`card` | CardPaymentSourceInput! | The card payment source information.
```

File: src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+
+### `CardPaymentSourceInput` object
+
+The `CardPaymentSourceInput` object describes the card payment source information.
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`billing_address` | [BillingAddressPaymentSourceInput]! | The billing address of the card.
```
Comment (keharper) [inline src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md:105]:
```suggestion
`billing_address` | BillingAddressPaymentSourceInput! | The billing address of the card.
```

File: src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+`address_line_1` | String | The first line of the address.
+`address_line_2` | String | The second line of the address.
+`region` | String | The region of the address.
+`city` | String | The city of the address.
+`postal_code` | String | The postal code of the address.
+`country_code` | String! | The country code of the address.
+
+## Output attributes
+
+The `createVaultCardSetupTokenOutput` object must contain the following attributes:
```
Comment (keharper) [inline src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md:123]:
```suggestion
The `createVaultCardSetupTokenOutput` object contains the following attribute:
```

File: src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+The `CardPaymentSourceInput` object describes the card payment source information.
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`billing_address` | [BillingAddressPaymentSourceInput]! | The billing address of the card.
+`name` | String! | The cardholder's name.
+
+### `BillingAddressPaymentSourceInput` object
+
+The `BillingAddressPaymentSourceInput` object includes the billing address information. Requires the following attributes:
```
Comment (keharper) [inline src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md:110]:
```suggestion
The `BillingAddressPaymentSourceInput` object includes the billing address information
```

File: src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+`postal_code` | String | The postal code of the address.
+`country_code` | String! | The country code of the address.
+
+## Output attributes
+
+The `createVaultCardSetupTokenOutput` object must contain the following attributes:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`setup_token` | String! | The setup token id.
```
Comment (keharper) [inline src/pages/graphql/schema/vault/mutations/createVaultCardSetupToken.md:127]:
```suggestion
`setup_token` | String! | The setup token ID
```

File: src/pages/graphql/schema/vault/queries/getVaultConfig.md [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+                        "value": "3600"
+                    }
+                ]
+            }
+        }
+    }
+}
+```
+
+For now, only `credit_card` is supported with the `getVaultConfig` query.
```
Comment (keharper) [inline src/pages/graphql/schema/vault/queries/getVaultConfig.md:75]: Delete

File: src/pages/graphql/schema/vault/queries/getVaultConfig.md [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+
+## Output attributes
+
+The `VaultConfigOutput` object must contain the following attribute:
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`credit_card` | [VaultCreditCardConfig] | Credit card vault method configuration.
```
Comment (keharper) [inline src/pages/graphql/schema/vault/queries/getVaultConfig.md:83]:
```suggestion
`credit_card` | VaultCreditCardConfig | Credit card vault method configuration.
```

File: src/pages/graphql/schema/vault/queries/getVaultConfig.md [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+        }
+    }
+}
+```
+
+For now, only `credit_card` is supported with the `getVaultConfig` query.
+
+## Output attributes
+
+The `VaultConfigOutput` object must contain the following attribute:
```
Comment (keharper) [inline src/pages/graphql/schema/vault/queries/getVaultConfig.md:79]:
```suggestion
The `VaultConfigOutput` object contains the following attribute:
```

File: src/pages/graphql/schema/vault/queries/getVaultConfig.md [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+`credit_card` | [VaultCreditCardConfig] | Credit card vault method configuration.
+
+### `VaultCreditCardConfig` attributes
+
+The `VaultCreditCardConfig` object provides details about the credit card vault method configuration
+
+Attribute |  Data Type | Description
+--- | --- | ---
+`is_vault_enabled` | Boolean | Indicates whether card vaulting is enabled.
+`three_ds_mode` | Boolean | Indicates which 3D Secure authentication mode is in use. The possible values are `OFF`, `SCA_WHEN_REQUIRED`, `SCA_ALWAYS`.
```
Comment (keharper) [inline src/pages/graphql/schema/vault/queries/getVaultConfig.md:92]:
```suggestion
`three_ds_mode` | ThreeDSMode | Indicates which 3D Secure authentication mode is in use. The possible values are `OFF`, `SCA_WHEN_REQUIRED`, `SCA_ALWAYS`
```

File: src/pages/graphql/schema/vault/queries/getVaultConfig.md [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+`sdk_params` | SDKParams | PayPal parameters required to load the PayPal JavaScript SDK.
```
Comment (keharper) [inline src/pages/graphql/schema/vault/queries/getVaultConfig.md:93]:
```suggestion
`sdk_params` | [SDKParams] | PayPal parameters required to load the PayPal JavaScript SDK
```

File: src/pages/graphql/schema/vault/index.md [hunk 20]
```diff
@@ -0,0 +1,15 @@
+---
+title: Vault
+---
+
+# Vault
+
+<InlineAlert variant="info" slots="text" />
+
```
[the remark this answers is not in the document]
Comment (danidelcar) [reply src/pages/graphql/schema/vault/index.md:8]: the issue I see with this is that vault was available before, but now with PS v2.10.0 there are more features, related to the new query and mutations available

File: src/pages/graphql/schema/vault/index.md [hunk 21]
```diff
@@ -0,0 +1,15 @@
+---
```
[the remark this answers is not in the document]
Comment (danidelcar) [reply src/pages/graphql/schema/vault/index.md:1]: I don't understand why we would move mutations and queries related to the checkout process to this folder. Vault existed before, now simply it has more features, thus new query and mutations, that are related to vaulting without a purchase, that is, outside of a checkout process, see [credit card vaulting in PS](https://experienceleague.adobe.com/en/docs/commerce-merchant-services/payment-services/payments-checkout/vaulting)

File: src/pages/graphql/schema/vault/queries/getVaultConfig.md (hunk 15 above)
Comment (danidelcar) [reply src/pages/graphql/schema/vault/queries/getVaultConfig.md:75]: why we want to delete this?. Right now, feature only works with `credit_card`

Comment (danidelcar) [reply src/pages/graphql/schema/vault/queries/getVaultConfig.md:75]: silly me, I already mentioned earlier in the same page, thanks!

**Comment by danidelcar** [comment]
Applied most of the suggestions @keharper , but I have a couple of questions regarding TOC and structure. Please take a look when you can, thanks!

File: src/pages/graphql/schema/vault/index.md (hunk 21 above)
[the remark this answers is not in the document]
Comment (danidelcar) [reply src/pages/graphql/schema/vault/index.md]: Applied a huge restructure for the payment services extension, might be the only thing needed is redirects

**Comment by danidelcar** [comment]
Please @virfaure @keharper take a look at the changes in the PR, thanks!

File: src/pages/graphql/payment-services-extension/index.md [hunk 22]
```diff
@@ -17,10 +17,24 @@ It provides powerful tools to give you the insight and control needed for creati
...
 * Apple Pay with PayPal
 * Google Pay with PayPal
 * PayPal Hosted Fields
 * PayPal Smart Buttons
-* PayPal Vault
+* Credit card vaulting
+
+## Credit card vaulting
+
+The vault-related changes to the schema allow you to generate stored payment details and to modify a payment tokens.
```
Comment (danidelcar) [inline src/pages/graphql/payment-services-extension/index.md:32]:
```suggestion
The vault-related changes to the schema allow you to generate stored payment details and to modify a payment token.
```

File: src/pages/graphql/payment-services-extension/index.md [hunk 23]
```diff
@@ -17,10 +17,24 @@ It provides powerful tools to give you the insight and control needed for creati
...
+
+## Credit card vaulting
+
+The vault-related changes to the schema allow you to generate stored payment details and to modify a payment token.
+
+The Payment Services Vault method is a solution in which shoppers can store—or "vault"—their credit card credentials to use in a later purchase:
+
+* Logged-in customers can vault a payment method in the **My Account** dashboard [without a purchase](https://experienceleague.adobe.com/en/docs/commerce-merchant-services/payment-services/payments-checkout/vaulting#vaulting-without-purchase). This functionality is available only if you have installed [Payment Services for Adobe Commerce](https://commercemarketplace.adobe.com/magento-payment-services.html) 2.10.0 or higher.
+
+* Logged-in customers can [vault a credit card during checkout](https://experienceleague.adobe.com/en/docs/commerce-merchant-services/payment-services/payments-checkout/vaulting#vaulting-a-payment-method-during-checkout) to use in a later purchases in the current store or other stores within the same merchant account.
```
Comment (danidelcar) [inline src/pages/graphql/payment-services-extension/index.md:38]:
```suggestion
* Logged-in customers can [vault a credit card during checkout](https://experienceleague.adobe.com/en/docs/commerce-merchant-services/payment-services/payments-checkout/vaulting#vaulting-a-payment-method-during-checkout) to use in a later purchase in the current store or other stores within the same merchant account.
```

**Review by virfaure — COMMENTED** [review]
👍 Good job

**Review by keharper — APPROVED** [review]

**Comment by danidelcar** [comment]
merged!, thanks @virfaure & @keharper for the help and reviews!

**Pull request merged**